src/app/public/src/modules/clipboard/clipboard.directive.ts

```diff
@@ -0,0 +1,80 @@
+// adapted from https://github.com/maxisam/ngx-clipboard/
+import {
+  Directive,
+  EventEmitter,
+  OnDestroy,
+  Output,
+  Renderer2,
+  ElementRef,
+  AfterViewInit,
+  ChangeDetectorRef,
+  Input
+} from '@angular/core';
+
+import { StacheClipboardService } from './clipboard.service';
+
+@Directive({
+  selector: '[stacheClipboard]'
+})
+export class StacheClipboardDirective implements AfterViewInit, OnDestroy {
+  // --------------------------------------------------------------
+  // NOTE: Only supported for Code-Block component as of July 2018
+  // --------------------------------------------------------------
+
+  private defaultTigger = 'clipboardButton';
+
+  @Input() public clipboardTrigger: string = this.defaultTigger;
```

Comment:
I added a comment about Template Reference Variables below, but this should be a `ElementRef` property.

See: https://angular.io/guide/template-syntax#ref-vars